src/sentry/db/models/outboxes.py

```diff
@@ -67,12 +81,83 @@ def outbox_for_update(self, shard_identifier: int | None = None) -> RegionOutbox
         raise NotImplementedError
 
 
+_RM = TypeVar("_RM", bound=RegionOutboxProducingModel)
+
+
+class RegionOutboxProducingManager(BaseManager[_RM]):
```

Comment:
Are there tests for the `bulk_*` methods here and on the control side?

Reply:
There are implicitly, because many tests fail thanks to mark's test helper methods that fail if you create models without outboxes anyways.  I can add some additional tests then to validate this behavior more deeply, yes.